### ISSUE TITLE
Panic: Runtime error: invalid memory address or nil pointer derefence

### ISSUE THREAD

**Issue by qzio** [docker/docker#22901]
During the night, docker crashed alongside all running containers.

The containers had exited `Exit(128)` and some with `Exit(65536)`

Docker daemon seemed to have restarted, but it was unable to restart the containers.

I removed them and was able to start new containers.

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x30 pc=0x87d60d]

[docker-crash.txt](https://github.com/docker/docker/files/277217/docker-crash.txt)

**Output of `docker version`:**

```
docker version
Client:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.5.4

 Git commit:   4dc5990
 Built:        Wed Apr 13 18:43:40 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   4dc5990
 Built:        Wed Apr 13 18:43:40 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

This was run after I removed the containers (since they were in a un-bootable state), and started new ones.

```
Containers: 20
 Running: 13
 Paused: 0
 Stopped: 7
Images: 14
Server Version: 1.11.0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 134
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host overlay bridge null
Kernel Version: 4.4.0-21-generic
Operating System: Ubuntu 16.04 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 15.67 GiB
Name: myhost
ID: P4XF:TG5B:DAJS:VPPJ:Z74Q:ZVBW:3QAP:RJXY:W4OX:FVWA:NZEO:PQVE
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Cluster store: consul://127.0.0.1:8500
Cluster advertise: 192.168.0.2:0
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

- Virtualized environment

**Steps to reproduce the issue:**
1. run containers
2. ?????
3. fail

**Describe the results you received:**

all containers stopped, and could not be started.

**Describe the results you expected:**

I did not expect the docker daemon to crash, and I did not expect the containers to be in a non-bootable state.

**Additional information you deem important (e.g. issue happens only occasionally):**

I have not been able to reproduce this.

**Comment by qzio** [comment]
Hmm. I wonder if this is a duplicate of https://github.com/docker/docker/issues/22081 ? if so, please close

**Comment by runcom** [comment]
@qzio it is, thx though, it's reported fixed in 1.11.1
